tokio/src/sync/semaphore.rs

```diff
@@ -186,7 +193,7 @@ impl Semaphore {
 
     /// Adds `n` new permits to the semaphore.
     ///
-    /// The maximum number of permits is `usize::MAX >> 3`, and this function will panic if the limit is exceeded.
+    /// The maximum number of permits is [`Semaphore::MAX_PERMITS`], and this function will panic if the limit is exceeded.
```

Comment:
Would be nice to keep the note that the value is `usize::MAX >> 3` somewhere in the docs, though probably not here.

Reply:
Isn't specific number visible in the docs (it looks like this: `pub const MAX_PERMITS: usize = 2_305_843_009_213_693_951usize`) clarify this?

Or do you mean the documentation should state that this number is fixed and is not expected to change?

Reply:
No, I mean that the docs should mention that `2_305_843_009_213_693_951usize` is equal to `usize::MAX >> 3`.

Reply:
Added sentence about that.

Also searched for other `usize::MAX >> 3` in the docs and replaced with a link to `MAX_PERMITS`.

Reply:
Another thing it would make sense to point out is that `2_305_843_009_213_693_951usize` is only the correct number on 64-bit platforms.

Reply:
`usize::MAX >> 3` implies different value on different platforms.